### ISSUE TITLE
Permalink - Problem with background layer

### ISSUE THREAD

**Issue by gnerred** [camptocamp/ngeo#2714]
The selected background is not kept when opening a permalink generated from NGEO mobile, version 2.1 or 2.2. To reproduce, open the following URL : 

https://map.cartoriviera.ch/mobile/theme/cadastre?lang=fr&tree_group_layers_mobilite=points_vente_vmcv%2Ctp_arret_bus%2Ctp_ligne_bus%2Ctp_arret_train%2Ctp_ligne_train&baselayer_ref=orthos_2014_mobile&map_x=555433&map_y=145317&map_zoom=9&theme=cadastre&tree_group_layers_amenagement_du_territoire=camac%2Ccamac_termine&tree_group_layers_cadastre=batiments_egid&tree_group_layers_couverture_du_sol&tree_groups=cadastre%2Ccouverture_du_sol

As you can see in above URL, baselayer_ref=orthos_2014_mobile . When opening the permalink, this parameter is automatically replaced with plan_cadastral, so the plan cadastral is shown instead of the ortho

**Comment by llienher** [comment]
I'm unfortunalty unable to reproduce in a recent project I work on using 2.2-RC3 or the gmf demos (2.1 and 2.2).

From what I saw debuging from the url provided above, in the source code that manage the permalink update, the baselayer_ref is instanciate correctly with orthos_2014_mobile then it is overwritten with the plan_cadastral value at the next loop iteration.

@gnerred 
I need to look further the project config and probably as well check what was set in the administration. Is it possible to provide me username/password for the preprod server admin ? (you can send it to my email: laurent.lienher@camptocamp.com)

**Comment by gnerred** [comment]
Yes, the "admin" account has a password already known and used by C2C. You can ask @ochriste or @sbrunner for that password.

**Comment by llienher** [comment]
@gnerred I did looked the config and this is my understanding of the current behaviour with ngeo (I looked your preprod instance for my tests).

In your project, each themes have a "functionality" attached to it. More specifically each theme use a "default_basemap" functionality. Some of the themes refering to the "plan_cadastral" default_basemap are loading this background layer as such, and it is the correct behaviour. It will only works if there is this specific background layer as well in the "background" layer group. 

Other themes using default_basemap are actually using values that are not in the "background" layer group, so because not found, it will not load their default background layers as excepted.

Basically this parameter set the theme to load a specific background layer when we active it (either by switching between themes or loading the web app).

So now, two possiblities:
1. You need that all the themes load a background layer when activing it, so you should change the references to an existing background layer used in your "background" layer group.
2. You don't want to use this, it should be disabled, unchecking the box about the "default_basemap" functionnality on each themes that are currently using it.

**Comment by llienher** [comment]
Now rethinking about all this, I assume the values you have in your administration are used by CGXP (the desktop interface). But, ngeo itself cannot know that.

**Comment by gnerred** [comment]
Sorry, but I'm not sure to undestand all what you mean... 

At this time, we have theme "Cadastre". that has a default basemap "Plan cadastral". When we go to theme "Cadastre", the default basemap is automatically set to "Plan cadastral". This works fine in NGEO and in CGXP. But, if we create a permalink in the theme "Cadastre" with another backgroundlayer selected, the permalink has to show this other backgroundlayer once opened. This work fine in CGXP. For example, here is a permalink of theme "Cadastre" with another backgroundlayer selected : 

https://preprod.cartoriviera.ch/s/piv0 
 
We expect to have the same behaviour in NGEO, and this is not the case. Because when we create a permalink in theme "Cadastre" in NGEO with another backgroundayer selected, it is allways the backgroundlyer "Plan cadastral" that is shown, and this is not the correct behaviour : 

https://preprod.cartoriviera.ch/mobile/theme/cadastre?lang=fr&tree_group_layers_mobilite=points_vente_vmcv%2Ctp_arret_bus%2Ctp_ligne_bus%2Ctp_arret_train%2Ctp_ligne_train&baselayer_ref=orthos_2014_mobile&map_x=555433&map_y=145317&map_zoom=9&theme=cadastre&tree_group_layers_amenagement_du_territoire=camac%2Ccamac_termine&tree_group_layers_cadastre=batiments_egid&tree_group_layers_couverture_du_sol&tree_groups=cadastre%2Ccouverture_du_sol

**Comment by ochriste** [comment]
@gnerred 
currently the behavior is: if a default default_basemap is specified in a theme (and, for the ngeo/mobile: that basemap exists in the "background" layergroup), that default_basemap will ALWAYS be the one activated.

so, practically, for all themes where you have defied a default_basemap and that basemap exists in the "background" layergroup, you will have the problem where the basemap passed in the url will be ignored.

**Comment by gnerred** [comment]
Ok, but this is not the expected behaviour... When we do a permalink, we expect to get the same map that was displayed when the permalink was generated. This works fine in CGXP, and I think that this has to be fixed in NGEO. Ping @ybolognini @rbovard @kalbermattenm what do you think ??

**Comment by rbovard** [comment]
Yes, it's a regression. It should be addressed ASAP.

**Comment by ger-benjamin** [comment]
> When we do a permalink, we expect to get the same map that was displayed when the permalink was generated. 

Exact. So that's a bug in ngeo. We'll work on that.

(Not part of this bug:
llienher has understood other things. It seems that when we set a theme in cartoriviera. It doesn't set the background. That's because some of your theme is linked to background layer values that not exist as background layer in ngeo. Exemple: The theme "sport" use the default basemap "asitvd.fond_couleur" that doesn't exist as background layer in your ngeo interface (look into "background" layergroup))

**Comment by gnerred** [comment]
Ok thanks @ger-benjamin .
